### PR TITLE
Clear all the queues before every test

### PR DESCRIPTION
**Pull request by Scooletz**
Connects to #60
solving it by clearing all the queues before every test

### CODE REVIEW

**Comment by Scooletz** [comment]
Ready to be merged.
Ping @Particular/azure-maintainers @dvdstelt

File: src/AcceptanceTests/ConfigureAzureStorageQueueTransport.cs [hunk 1]
```diff
@@ -45,28 +41,28 @@ public Task Cleanup()
...
         {
-            var queue = queues.GetQueueReference(queuesName);
-            if (await queue.ExistsAsync())
-            {
-                await queue.ClearAsync();
-            }
+            queue.ClearAsync().ContinueWith(t => countdown.Signal());
+        }
+
+        if (countdown.Wait(TimeSpan.FromMinutes(1)) == false)
```
Comment (dvdstelt) [inline src/AcceptanceTests/ConfigureAzureStorageQueueTransport.cs]: Nice solution!

**Comment by SeanFeldman** [comment]
Purging all queues approach is going to be challenging as `IConfigureEndpointTestExecution` methods (`Configure` and `Cleanup`) are involved per each endpoint participating in acceptance test. That means we'll clean all the queues N times (N is the number of endpoints per ATT).

Don't know if NUnit can indicate when the whole test suite is starting or ending. If that would be possible, it would be an ideal place to hook into.

**Comment by Scooletz** [comment]
@SeanFeldman The one way of doing is providing `ITestAction` http://nunit.org/index.php?p=actionAttributes&r=2.6.2 but this requires applying it on all the suites. The other approach is using the provided extensibility point & by writing a small plugin implementing http://nunit.org/index.php?p=testDecorators&r=2.6.2 run a piece of code before every suite. Time: up to 1 day.

**Pull request merged**